### PR TITLE
[3.12.x] Add client full example configuration files

### PR DESCRIPTION
**Pull request by blazember**
There were `hazelcast-client-full.[xml|yaml]` configuration files shipped with the `hazelcast-client.jar`. These files however had no comments, and the files were used by tests too. To make test files and full example configs separated, these files are moved to the test scope, but also used as a basis for the full example config files that are extended with explanatory comments. Client failover full example configs are added too.

Besides this, the client failover config XSDs got fixed. These fixes include:
- Defining target namespace
- Change the `<try-count>` element's type from `xs:int` to `non-space-string` otherwise schema validation error is thrown if the variable replacer is used to configure the try count value.

The client failover XSD is changed retrospectively, but without the fix, schema validation fails if used. This needs further coordination to update the XSD.

Implements #14948

4.0 PR: https://github.com/hazelcast/hazelcast/pull/15056

3.12.x EE PR: hazelcast/hazelcast-enterprise#2992

### CODE REVIEW

File: hazelcast-client/src/main/resources/hazelcast-client-failover-config-3.12.xsd [hunk 1]
```diff
@@ -16,13 +16,15 @@
...
+           xmlns="http://www.hazelcast.com/schema/client-config"
            elementFormDefault="qualified"
+           targetNamespace="http://www.hazelcast.com/schema/client-config"
            attributeFormDefault="unqualified">
 
     <xs:element name="hazelcast-client-failover">
         <xs:complexType>
             <xs:choice minOccurs="1" maxOccurs="unbounded">
-                <xs:element name="try-count" type="xs:int" minOccurs="1" maxOccurs="1"/>
+                <xs:element name="try-count" type="non-space-string" minOccurs="1" maxOccurs="1"/>
```
Comment (sancar) [inline hazelcast-client/src/main/resources/hazelcast-client-failover-config-3.12.xsd]: So, we should not use xs:int anywhere to support replacers, is that the case?
I have just checked we have usages of it in hazelcast-client-config-3.12.xsd. Should all be removed?

Comment (blazember) [reply hazelcast-client/src/main/resources/hazelcast-client-failover-config-3.12.xsd]: I revert this part of the schema change. As you said, this is a general problem with replacers. We have a solution for this in the spring schema: https://github.com/hazelcast/hazelcast/blob/d47c1b5962146de2711fa0e4950fa0cfa91b7083/hazelcast-spring/src/main/resources/hazelcast-spring-3.12.xsd#L3368-L3370
I believe this approach should be used by the other XSDs as well, but this needs no retrospective change.

However, I still think adding `xmlns` and `targetNamespace` needs retrospective change.

File: hazelcast-client/src/main/resources/hazelcast-client-full-example.xml [hunk 2]
```diff
@@ -0,0 +1,879 @@
...
+        When Hazelcast instances are created, they are put in a global registry with their creation names.
+        <instance-name> elements gives you the ability to get a specific Hazelcast instance from this registry
+        by giving the instance's name.
+    -->
+    <instance-name>CLIENT_NAME</instance-name>
+
+    <!--
+        ===== HAZELCAST LICENSE CONFIGURATION =====
+
+        If you have an Enterprise or Enterprise HD license, you can enter it here.
```
Comment (asimarslan) [inline hazelcast-client/src/main/resources/hazelcast-client-full-example.xml]: On the client, licence is not required. This config is for backward compatibility. We may update the comment to imply that it's not required.

**Comment by blazember** [comment]
Thanks for the review @asimarslan @sancar :+1:

**Pull request merged**